### ISSUE TITLE
Specifying both of host and build option in configure causes configure error

### ISSUE THREAD

**Issue by cosmo0920** [julius-speech/julius#77]
This issue is caused by Yocto's poky build system.

Yocto's poky specifies like the following configure option in `oe_runconf()`:

```text
CONFIGUREOPTS = " --build=${BUILD_SYS} \
		  --host=${HOST_SYS} \
		  --target=${TARGET_SYS} \
		  --prefix=${prefix} \
		  --exec_prefix=${exec_prefix} \
		  --bindir=${bindir} \
		  --sbindir=${sbindir} \
		  --libexecdir=${libexecdir} \
		  --datadir=${datadir} \
		  --sysconfdir=${sysconfdir} \
		  --sharedstatedir=${sharedstatedir} \
		  --localstatedir=${localstatedir} \
		  --libdir=${libdir} \
		  --includedir=${includedir} \
		  --oldincludedir=${oldincludedir} \
		  --infodir=${infodir} \
		  --mandir=${mandir} \
		  --disable-silent-rules \
                  # other arguments are poky internal.
```

 Minimum reproducible step on Debian Stretch:

```bash
 ./configure --build=x86_64-linux-gnu --host=aarch64-linux-gnu # or using --host=arm-linux-gnueabihf
```

will cause configure failure:

```log
% ./configure --build=x86_64-linux-gnu --host=aarch64-linux-gnu
checking build system type... x86_64-pc-linux-gnu
checking host system type... aarch64-unknown-linux-gnu
checking host specific optimization flag... no
checking for aarch64-linux-gnu-gcc... aarch64-linux-gnu-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether aarch64-linux-gnu-gcc accepts -g... yes
checking for aarch64-linux-gnu-gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... aarch64-linux-gnu-gcc -E
checking for a BSD-compatible install... /usr/bin/install -c
checking for rm... /bin/rm
configure: creating ./config.status
config.status: creating Makefile
config.status: creating mkbinhmm/Makefile
config.status: creating adinrec/Makefile
config.status: creating mkss/Makefile
config.status: creating generate-ngram/Makefile
config.status: creating jclient-perl/Makefile
config.status: creating binlm2arpa/Makefile
config.status: creating man/Makefile
config.status: WARNING:  'man/Makefile.in' seems to ignore the --datarootdir setting
=== configuring in adintool (/home/hhatake/GitHub/julius/adintool)
configure: running /bin/bash ./configure --disable-option-checking '--prefix=/usr/local'  '--build=x86_64-linux-gnu' '--host=aarch64-linux-gnu' 'build_alias=x86_64-linux-gnu' 'host_alias=aarch64-linux-gnu' --cache-file=/dev/null --srcdir=.
checking build system type... x86_64-pc-linux-gnu
checking host system type... aarch64-unknown-linux-gnu
checking host-specific optimization flag... no
checking for aarch64-linux-gnu-gcc... aarch64-linux-gnu-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether aarch64-linux-gnu-gcc accepts -g... yes
checking for aarch64-linux-gnu-gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... aarch64-linux-gnu-gcc -E
checking for a BSD-compatible install... /usr/bin/install -c
checking for rm... /bin/rm
checking for aarch64-linux-gnu-pkg-config... no
checking for pkg-config... /usr/bin/pkg-config
configure: WARNING: using cross tools not prefixed with host triplet
checking pkg-config is at least version 0.9.0... yes
checking for SDL... no
checking for sdl2-config... no
checking for SDL - version >= 2.0.0... no
*** The sdl2-config script installed by SDL could not be found
*** If SDL was installed in PREFIX, make sure PREFIX/bin is in
*** your path, or set the SDL2_CONFIG environment variable to the
*** full path to sdl2-config.
configure: creating ./config.status
config.status: creating Makefile
=== configuring in mkgshmm (/home/hhatake/GitHub/julius/mkgshmm)
configure: running /bin/bash ./configure --disable-option-checking '--prefix=/usr/local'  '--build=x86_64-linux-gnu' '--host=aarch64-linux-gnu' 'build_alias=x86_64-linux-gnu' 'host_alias=aarch64-linux-gnu' --cache-file=/dev/null --srcdir=.
configure: warning: build_alias=x86_64-linux-gnu: invalid host type
configure: warning: host_alias=aarch64-linux-gnu: invalid host type
configure: error: can only configure for one host and one target at a time
configure: error: ./configure failed for mkgshmm
```

**Comment by nitslp-ri** [comment]
Is this error still occurs?  What happens if you reproduce the configure scripts in the sub-directories by recent autoconf?

**Comment by cosmo0920** [comment]
Yes. This error still occurs at `mkgshmm`. 

autoreconf results:

#### toplevel

```
% autoreconf
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
configure.in:42: warning: macro 'AM_PATH_SDL2' not found in library
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
configure.in:63: warning: macro 'AM_PATH_GTK' not found in library
configure.in:63: error: possibly undefined macro: AM_PATH_GTK
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /usr/bin/autoconf failed with exit status: 1
```

#### mkgshmm

```
% autoreconf
aclocal: warning: autoconf input should be named 'configure.ac', not 'configure.in'
```

**Comment by nitslp-ri** [comment]
OK, I've reproduced the error on my pc, and confirmed regenerating configure script with autoconf-2.69 fixes the problem.  The fix has been commited, so could you try the latest code?

34640dd93e0b957741743dea59b696349edce76c

**Comment by cosmo0920** [comment]
I also confirmed that ./configure can pass with https://github.com/julius-speech/julius/commit/34640dd93e0b957741743dea59b696349edce76c.
I've got compilation error due to corss compiled dependent libraries missing.
This is another issue for me and https://github.com/julius-speech/julius/commit/34640dd93e0b957741743dea59b696349edce76c is enough to close this issue.
Thanks for fixing this issue. :grin:
